File: 1.Domain/Domain.Model/HrStage.cs
using Core;
using Domain.Model.Enum;
using System;

namespace Domain.Model
{
    public class HrStage : Entity<int>
    {
        public StageType Type { get; set; }
        public DateTime? Date { get; set; }
        public string Feedback { get; set; }
        public StageStatus Status { get; set; }
        public int ProcessId { get; set; }
        public Process Process { get; set; }
        public int? UserOwnerId { get; set; }
        public User UserOwner { get; set; }
        public int? UserDelegateId { get; set; }
        public User UserDelegate { get; set; }
        public string RejectionReason { get; set; }
        public float ActualSalary { get; set; }
        public float WantedSalary { get; set; }
        public string AdditionalInformation { get; set; }
        public EnglishLevel EnglishLevel { get; set; }
        public RejectionReasonsHr RejectionReasonsHr { get; set; }
        public bool SentEmail { get; set; }
        public int? ReaddressStatusId { get; set; }
        public ReaddressStatus ReaddressStatus { get; set; }
    }
}

File: 4.Clients/WebApi/ClientApp/src/entities/hr-stage.ts
import { StageStatusEnum } from './enums/stage-status.enum';
import { EnglishLevelEnum } from './enums/english-level.enum';
import { RejectionReasonsHrEnum } from './enums/rejection-reasons-hr.enum';

export class HrStage {
    id: number;
    date: Date;
    feedback: string;
    status: StageStatusEnum;
    processId: number;
    actualSalary?: number;
    wantedSalary?: number;
    additionalInformation: string;
    englishLevel: EnglishLevelEnum;
    rejectionReasonsHr?: RejectionReasonsHrEnum;
    userOwnerId?: number;
    userDelegateId?: number;
    rejectionReason?: string;
    sentEmail: boolean;
}
